steering_controllers/include/steering_controllers/steering_controllers.hpp

```diff
@@ -0,0 +1,143 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef STEERING_CONTROLLERS__STEERING_CONTROLLERS_HPP_
+#define STEERING_CONTROLLERS__STEERING_CONTROLLERS_HPP_
+
+#include <chrono>
+#include <cmath>
+#include <memory>
+#include <queue>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "controller_interface/chainable_controller_interface.hpp"
+#include "hardware_interface/handle.hpp"
+#include "rclcpp_lifecycle/node_interfaces/lifecycle_node_interface.hpp"
+#include "rclcpp_lifecycle/state.hpp"
+#include "realtime_tools/realtime_buffer.h"
+#include "realtime_tools/realtime_publisher.h"
+#include "std_srvs/srv/set_bool.hpp"
+#include "steering_controllers/steering_odometry.hpp"
+#include "steering_controllers/visibility_control.h"
+#include "steering_controllers_parameters.hpp"
+
+// TODO(anyone): Replace with controller specific messages
+#include "ackermann_msgs/msg/ackermann_drive.hpp"
+#include "control_msgs/msg/steering_controller_status.hpp"
+#include "geometry_msgs/msg/twist.hpp"
+#include "geometry_msgs/msg/twist_stamped.hpp"
+#include "nav_msgs/msg/odometry.hpp"
+#include "tf2_msgs/msg/tf_message.hpp"
+
+namespace steering_controllers
+{
+class SteeringControllers : public controller_interface::ChainableControllerInterface
+{
+public:
+  STEERING_CONTROLLERS__VISIBILITY_PUBLIC SteeringControllers();
+
+  STEERING_CONTROLLERS__VISIBILITY_PUBLIC controller_interface::CallbackReturn on_init() override;
+
+  STEERING_CONTROLLERS__VISIBILITY_PUBLIC controller_interface::InterfaceConfiguration
+  command_interface_configuration() const override;
+
+  STEERING_CONTROLLERS__VISIBILITY_PUBLIC controller_interface::InterfaceConfiguration
+  state_interface_configuration() const override;
+
+  virtual STEERING_CONTROLLERS__VISIBILITY_PUBLIC controller_interface::CallbackReturn
+  configure_odometry();
+
+  virtual STEERING_CONTROLLERS__VISIBILITY_PUBLIC bool update_odometry(
+    const rclcpp::Duration & period);
+
+  STEERING_CONTROLLERS__VISIBILITY_PUBLIC controller_interface::CallbackReturn on_configure(
+    const rclcpp_lifecycle::State & previous_state) override;
+
+  STEERING_CONTROLLERS__VISIBILITY_PUBLIC controller_interface::CallbackReturn on_activate(
+    const rclcpp_lifecycle::State & previous_state) override;
+
+  STEERING_CONTROLLERS__VISIBILITY_PUBLIC controller_interface::CallbackReturn on_deactivate(
+    const rclcpp_lifecycle::State & previous_state) override;
+
+  STEERING_CONTROLLERS__VISIBILITY_PUBLIC controller_interface::return_type
+  update_reference_from_subscribers() override;
+
+  STEERING_CONTROLLERS__VISIBILITY_PUBLIC controller_interface::return_type
+  update_and_write_commands(const rclcpp::Time & time, const rclcpp::Duration & period) override;
+
+  // TODO(anyone): replace the state and command message types
+  using ControllerReferenceMsg = geometry_msgs::msg::TwistStamped;
```

Comment:
Add also reference mssage for `ackermann_msgs` type
```suggestion
  using ControllerAckermannReferenceMsg = acermann_msgs::msg::AckermannDriveStamped;
  using ControllerTwistReferenceMsg = geometry_msgs::msg::TwistStamped;
```